src/DashboardService.Data/TaskRepository.cs

```diff
@@ -18,28 +21,114 @@ public TaskRepository(IDataProvider provider)
     _provider = provider;
   }
   
-  public Task<Guid?> CreateAsync(DbTask dbDepartment)
+  public async Task<Guid> CreateAsync(DbTask dbTask)
   {
-    throw new NotImplementedException();
+    _provider.Tasks.Add(dbTask);
+    await _provider.SaveAsync();
+
+    return dbTask.Id;
   }
 
-  public Task<List<DbTask>> GetAllAsync(GetTasksFilter filter)
+  public async Task<(List<DbTask> dbTasks, int totalCount)> GetAllAsync(GetTasksFilter filter, CancellationToken ct)
   {
-    throw new NotImplementedException();
+    IQueryable<DbTask> dbTasks = _provider.Tasks.Where(t => !filter.GroupId.HasValue || t.GroupId == filter.GroupId).AsNoTracking();
+
+    if (filter.BoardId.HasValue)
+    {
+      DbBoard dbBoard = await _provider.Boards
+        .Include(b => b.Groups)
+        .FirstOrDefaultAsync(b => b.Id == filter.BoardId, ct);
+      
+      dbTasks = dbTasks
+        .Include(t => t.Group)
+        .Where(t => dbBoard.Groups.FirstOrDefault(t.Group) != default);
+    }
+    
+    dbTasks = dbTasks.Where(t => !filter.PriorityId.HasValue || t.PriorityId == filter.PriorityId).AsNoTracking();
```

Comment:
Why do you need so many AsNoTracking()? You use it on 34 line.

Reply:
)) removed